tree_test.go

```diff
@@ -1278,3 +1325,90 @@ func BenchmarkBatchLeavesInsert(b *testing.B) {
 		}
 	}
 }
+
+func TestManipulateChildren(t *testing.T) {
+	root := New()
+	root.Insert(ffx32KeyTest, testValue, nil)
+
+	// Verify that Children() is returning what's expected.
+	ln, ok := root.(*InternalNode).Children()[NodeWidth-1].(*LeafNode)
+	if !ok {
+		t.Fatalf("failed to get expected leaf node")
+	}
+	if !bytes.Equal(ln.stem, ffx32KeyTest[:StemSize]) || !bytes.Equal(ln.values[NodeWidth-1], testValue) {
+		t.Fatalf("failed to get expected leaf node stem and values")
+	}
+
+	// Verify SetChild()
+	if err := root.(*InternalNode).SetChild(0, Empty{}); err != nil {
+		t.Fatalf("failed to set child: %v", err)
+	}
+	if _, ok := root.(*InternalNode).Children()[0].(Empty); !ok {
+		t.Fatalf("failed to set child")
+	}
+
+	// Verify SetChild() error case.
+	if err := root.(*InternalNode).SetChild(NodeWidth, Empty{}); err == nil {
+		t.Fatalf("expected error")
+	}
+}
+
+func TestLeafNodeInsert(t *testing.T) {
```

Comment:
Some LeafNode methods were uncovered (i.e: `Value(...)`, `Values(...)` again used in go-ethereum but not generally in go-verkle). Making test tighter to cover them here.